Packs/FeedCrowdstrikeFalconIntel/Integrations/CrowdStrikeIndicatorFeed/CrowdStrikeIndicatorFeed.py

```diff
@@ -132,77 +132,105 @@ def fetch_indicators(self, limit: Optional[int], offset: Optional[int] = 0, fetc
                                                  for item in self.malicious_confidence])
             filter = f"{filter}+({malicious_confidence_fql})" if filter else f'({malicious_confidence_fql})'
 
-        if fetch_command:
-            offset = demisto.getIntegrationContext().get('offset', 0)
-            demisto.info(f' current offset: {offset}')
-
         if manual_last_run:
             filter = f'{filter}+(last_updated:>={manual_last_run})' if filter else f'(last_updated:>={manual_last_run})'
 
         if fetch_command:
             if last_run := self.get_last_run():
                 filter = f'{filter}+({last_run})' if filter else f'({last_run})'
-            elif self.first_fetch:
-                last_run = f'last_updated:>={int(self.first_fetch)}'
-                filter = f'{filter}+({last_run})' if filter else f'({last_run})'
+            else:
+                filter, indicators = self.handling_first_fetch_and_old_integration_context(filter)
+                if indicators:
+                    limit = limit - len(indicators)
+
+        if filter or not fetch_command:
+            demisto.info(f'{filter=}')
+            params = assign_params(include_deleted=self.include_deleted,
+                                   limit=limit,
+                                   offset=offset, q=self.generic_phrase,
+                                   filter=filter,
+                                   sort='_marker|asc')
+
+            response = self.get_indicators(params=params)
+
+            # need to fetch all indicators after the limit
+            if resources := response.get('resources', []):
+                new_last_marker_time = resources[-1].get('_marker')
+            else:
+                new_last_marker_time = demisto.getIntegrationContext().get('last_marker_time')
+                demisto.debug('There are no indicators, using last_marker_time from Integration Context')
+
+            if fetch_command:
+                ctx = demisto.getIntegrationContext()
```

Comment:
```suggestion
                context = demisto.getIntegrationContext()
```
clearer names are preferred over shorter ones